array_methods_lab.js

```diff
@@ -0,0 +1,185 @@
+// console.log('-----QUESTION 1------')
+// /*iven an array of numbers, write a function that
+// returns a new array where every element is incremented by 10.*/
+//
+// let numArr = [1,2,3,4,5]
+// let addTen = numArr.map((crrElem) => {
+//   return crrElem +  10
+// })
+// console.log(addTen)
+
+
+
+
+// console.log('********QUESTION 2*********')
+// /*Given an array, return a new array where every
+// non-string element is converted to an empty string.*/
+//
+// let mixedArr = ["a", 123, "b", "c", {name: "cat"}]
+// let strArr = mixedArr.map((crrElem) => {
+//   if (typeof(mixedArr) === 'number') {
+//   return ""
+//   }
+//   return mixedArr
+// })
+// console.log(strArr)
+
+
+
+
+// console.log('((((((((((((QUESTION 3))))))))))))')
+// //Write a function that returns only the even elements in an array of numbers.
+//
+// let ranArr = [1,2,3,4,5]
+// let evenArr = ranArr.filter((crrElem) => {
+//   if (crrElem % 2 === 0) {
+//     return true
+//   }
+//   false
+// })
+// console.log(evenArr)
+
+
+
+
+// console.log('<<<<<<<<QUESTION 4>>>>>>>>>')
+// //Write a function that checks if all elements in the array are the same.
+// let sameNumArr = [2,2,2,2,2,2]
+// let dittoArr = sameNumArr.every(elem => {
+//   if (elem === elem) {
+//     return true
+//   }
+//   return false
+// })
+// console.log(dittoArr)
+//
```

Comment:
On the right path... So when I changed the array to [2,1,2,2,2,] --it still returned true... The reason that it returns true is because an elem will always be equal to themselves. Can you think of another way to compare the equality of each element within the array? We just have to change our condition and it'll be perfect.